scripts/Diffraction/isis_powder/abstract_inst.py

```diff
@@ -243,6 +243,23 @@ def _output_focused_ws(self, processed_spectra, run_details, output_mode=None):
 
         return d_spacing_group, tof_group
 
+    def create_solid_angle_corrections(self, vanadium, run_details):
+        """
+        Creates and saves the solid angle corrections from a vanadium run, only applicable on HRPD otherwise return None
+        :param vanadium: The vanadium used to create this
+        :param run_details: The run details used
+        """
+        return None
```

Comment:
Just a note here about abstract python classes. Using either abc's abstract method decorator (https://docs.python.org/3/library/abc.html) or `raise NotImplementedError`. That way you can ensure that you haven't accidentally forgotten to implement something.

I'm not sure if they are good solutions for user-exposed script, however. I'll leave that up to you